pebblo/entity_classifier/entity_classifier.py

```diff
@@ -43,13 +44,27 @@ def custom_analyze(self):
     def analyze_response(self, input_text, anonymize_all_entities=True):
         # Returns analyzed output
         analyzer_results = self.analyzer.analyze(text=input_text, language="en")
-        analyzer_results = [
-            result
-            for result in analyzer_results
-            if result.score >= float(ConfidenceScore.Entity.value)
-            and result.entity_type in self.entities
-        ]
-        return analyzer_results
+        final_results = []
+        for entity in analyzer_results:
+            try:
+                mapped_entity = None
+                if entity.entity_type in Entities.__members__:
+                    mapped_entity = Entities[entity.entity_type].value
+                elif entity.entity_type in SecretEntities.__members__:
+                    mapped_entity = SecretEntities[entity.entity_type].value
+                if (
+                    mapped_entity
+                    and entity.score
+                    >= float(entity_group_conf_mapping[mapped_entity][0])
+                    and entity.entity_type in self.entities
```

Comment:
One liner comment in the code would be good here stating that we are checking threshold for that particular entity.